cypress/integration/edit_image_mobile.spec.js

```diff
@@ -0,0 +1,55 @@
+context('Edit Image Mobile', () => {
+  let image;
+
+  beforeEach(() => {
+    cy.viewport('iphone-6');
+    cy.resetDB();
+    cy.authenticate();
+    cy.setActiveTeam()
+      .then((team) => {
+        cy.createImage('goat.jpg', team.id, ['Goat'])
+          .then((createdImage) => {
+            image = createdImage;
+            cy.visit(`/images/${image.id}/edit`);
+            cy.server();
+            cy.route({ method: 'PATCH', url: `/images/${image.id}` }).as('updateImage');
+          });
+      });
+  });
+
+  it('redirects to the detail page on successful submit', () => {
+    cy.get('[data-cy=save-button]').click();
+    cy.wait('@updateImage');
+    cy.location().should((location) => {
+      expect(location.pathname).to.eq(`/images/${image.id}`);
```

Comment:
such tests don't really require backend integration and shouldn't be tested in e2e imo. You could easily mock the server and check that it redirects in a unit test instead of starting the whole service suite for something like this.

E2E tests are expensively time consuming for everyone having them for core functionality is very valuable but testing details like a redirect with it costs too many resources imo.

Reply:
You are correct! Let's keep save e2e resources for big tests.